### DISCUSSION TITLE
Getting Exceptions when Doing Patrol Setup for IOS Native

### DISCUSSION THREAD

**Discussion by akagupta9** [leancodepl/patrol#1098] [Q&A]
<img width="1440" alt="Screenshot 2023-03-17 at 11 51 51 AM" src="https://user-images.githubusercontent.com/43808980/225828615-99e8f65a-9b7a-4c60-ab9d-d23cb92bad6a.png">

**Comment by bartekpacia** [comment]
You probably did something wrong in the native setup.

Please show the files you edited in every step of the native setup (`Podfile`,  `ios/RunnerUITests/RunnerUITests.m`)

It'd be best if you created [a minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of your  problem. I can't help just by looking at that screenshot.

**Comment by akagupta9** [comment]
`@import XCTest;
@import patrol;

PATROL_INTEGRATION_TEST_IOS_RUNNER(RunnerUITests)
`

This is my ios/RunnerUITests/RunnerUITests.m

`target 'Runner' do
  use_frameworks!
  use_modular_headers!
  pod 'GoogleUtilities'

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
  
  target 'RunnerUITests' do
    inherit! :complete
  end
end`

I have added that in podfile - Podfile

**Reply by bartekpacia** [reply]
Can you show the whole `Podfile`? Specifically, it looks like you got the order slightly wrong. It should be:

```ruby
target 'Runner' do
  use_frameworks!
  use_modular_headers!

  target 'RunnerUITests' do
    inherit! :complete
  end

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end
```

You can also take a look at our [example app](https://github.com/leancodepl/patrol/blob/master/packages/patrol/example/ios/Podfile) – feel free to clone it locally and run, to see how a correct setup looks like.

**Comment by akankshajify** [comment]
Order didnt helped.

```  # Uncomment this line to define a global platform for your project
platform :ios, '13.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!
  pod 'GoogleUtilities'
  
  target 'RunnerTests' do
    inherit! :search_paths
   end

  target 'RunnerUITests' do
    inherit! :complete
  end

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

target 'RichNotification' do
  use_frameworks!
  pod 'Firebase/Messaging'
  pod 'GoogleUtilities'
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
          config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
             '$(inherited)',

             ## dart: PermissionGroup.camera
             'PERMISSION_CAMERA=1',

             ## dart: PermissionGroup.photos
             'PERMISSION_PHOTOS=0',

             ## dart: [PermissionGroup.location, PermissionGroup.locationAlways, PermissionGroup.locationWhenInUse]
             'PERMISSION_LOCATION=1',

             ## dart: PermissionGroup.notification
             'PERMISSION_NOTIFICATIONS=1',

             ## dart: PermissionGroup.mediaLibrary
             'PERMISSION_MEDIA_LIBRARY=1',

             'PERMISSION_MICROPHONE=1',

           ]
    end
  end
end
```

**Comment by akankshajify** [comment]
It was not taking right flutter root path in build phase, which caused it.

**Reply by bartekpacia** [reply]
Thanks for sharing this! Can you explain in more detail how you solved it?

**Comment by akagupta9** [comment]
I put complete path for sh file in build phase `xcode_backend build`.

**Reply by bartekpacia** [reply]
If you need to do this, you have probably misconfigured the Build Configuration in Xcode. In a simple app without flavors, it should be set like this:


![Screenshot 2023-03-21 at 2 58 04 PM](https://user-images.githubusercontent.com/40357511/226628841-5f403ea2-6fce-418f-b5f1-d6d28355232d.png)
